loki/ir.py

```diff
@@ -941,6 +941,41 @@ def arg_map(self):
         """
         return dict(self.arg_iter())
 
+    def _sort_kwarguments(self):
+        """
+        Helper routine to sort the kwarguments according to the order of the
+        arguments (``self.routine.arguments``)`.
+        """
+        routine = self.routine
+        assert routine is not BasicType.DEFERRED
+        kwargs = CaseInsensitiveDict(self.kwarguments) # [kwarg[0] for kwarg in self.kwarguments]
+        r_arg_names = [arg.name for arg in routine.arguments if arg.name in kwargs]
+        new_kwarguments = tuple((arg_name, kwargs[arg_name]) for arg_name in r_arg_names)
+        return new_kwarguments
+
+    def check_kwarguments_order(self):
+        """
+        Check whether kwarguments are correctly ordered
+        in respect to the arguments (``self.routine.arguments``).
+        """
+        return self.kwarguments == self._sort_kwarguments()
+
+    def sort_kwarguments(self):
+        """
+        Sort and update the kwarguments according to the order of the
+        arguments (``self.routine.arguments``).
+        """
+        new_kwarguments = self._sort_kwarguments()
+        self._update(kwarguments=new_kwarguments)
+
+    def kwargs_to_args(self):
```

Comment:
Maybe use the "leading verb" naming here, too?

```suggestion
    def convert_kwargs_to_args(self):
```